java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java

```diff
@@ -429,17 +430,37 @@ public static ErrorDescription returnNull(HintContext ctx) {
 
         TreePath method = ctx.getPath();
 
-        while (method != null && method.getLeaf().getKind() != Kind.METHOD && method.getLeaf().getKind() != Kind.CLASS) {
+        loop: while (method != null) {
```

Comment:
Consider to use org.netbeans.modules.java.hints.errors.Utilities.findOwningExecutable

Reply:
Done